src/app/ir/conv.ts

```diff
@@ -0,0 +1,73 @@
+import { InOutNode, ModelNode } from '@/app/ir/mainNodes';
+import { Option, UUID } from '@/app/util';
+import {
+  ActivationF, Initializer, Padding, Regularizer,
+} from '@/app/ir/irCommon';
+
+abstract class Conv implements ModelNode, InOutNode {
+  abstract readonly outputs: Set<UUID>;
+
+  abstract readonly input: UUID;
+
+  abstract readonly filters: bigint;
+
+  abstract readonly padding: Padding;
+
+  abstract readonly weights: [Initializer, Regularizer]
+
+  abstract readonly biases: Option<[Initializer, Regularizer]>
+
+  abstract readonly activation: ActivationF
+}
+
+class Conv1D extends Conv {
+  constructor(
+    public readonly outputs: Set<UUID>,
+    public readonly filters: bigint,
+    public padding: Padding,
+    public weights: [Initializer, Regularizer],
+    public readonly biases: Option<[Initializer, Regularizer]>,
+    public readonly input: UUID,
+    public readonly dilation: [bigint],
```

Comment:
why bigint and not number for all params?

Reply:
to enforce whole numbers, we don't want a dilation of 2.3